src/celeritas/ext/detail/CelerOpticalPhysics.cc

```diff
@@ -0,0 +1,277 @@
+    auto absorption = std::make_unique<G4OpAbsorption>();
+    if (process_is_active("OpAbsorption"))
+    {
+        process_manager->AddDiscreteProcess(absorption.release());
+        CELER_LOG(debug) << "Loaded Optical absorption with G4OpAbsorption "
+                            "process";
+    }
+
+    auto rayleigh = std::make_unique<G4OpRayleigh>();
+    if (process_is_active("OpRayleigh"))
+    {
+        process_manager->AddDiscreteProcess(rayleigh.release());
+        CELER_LOG(debug)
+            << "Loaded Optical Rayleigh scattering with G4OpRayleigh "
+               "process";
+    }
+
+    auto mie = std::make_unique<G4OpMieHG>();
+    if (process_is_active("OpMieHG"))
+    {
+        process_manager->AddDiscreteProcess(mie.release());
+        CELER_LOG(debug) << "Loaded Optical Mie (Henyey-Greenstein phase "
+                            "function) scattering with G4OpMieHG "
+                            "process";
+    }
+
+    // NB: boundary is also used later on in loop over particles,
+    // though it's only ever applicable to G4OpticalPhotons
+    auto boundary = std::make_unique<G4OpBoundaryProcess>();
+#if G4VERSION_NUMBER < 1070
+    boundary->SetInvokeSD(options_.invoke_sd);
+#endif
+    if (process_is_active("OpBoundary"))
+    {
+        process_manager->AddDiscreteProcess(boundary.get());
+        CELER_LOG(debug)
+            << "Loaded Optical boundary process with G4OpBoundaryProcess "
+               "process";
+    }
+
+    auto wls = std::make_unique<G4OpWLS>();
+#if G4VERSION_NUMBER < 1070
+    wls->UseTimeProfile(to_cstring(options_.wavelength_shifting));
+#endif
+    if (process_is_active("OpWLS"))
+    {
+        process_manager->AddDiscreteProcess(wls.release());
+        CELER_LOG(debug) << "Loaded Optical wavelength shifting with G4OpWLS "
+                            "process";
+    }
+
+#if G4VERSION_NUMBER >= 1070
+    auto wls2 = std::make_unique<G4OpWLS2>();
+    if (process_is_active("OpWLS2"))
+    {
+        process_manager->AddDiscreteProcess(wls2.release());
+        // I need to check how this differs from G4OpWLS...
+        CELER_LOG(debug)
+            << "Loaded Optical wavelength shifting V2 with G4OpWLS2 "
+               "process";
+    }
+#endif
+
+    // Add photon-generating processes to all particles they apply to
+    // TODO: Eventually replace with Celeritas step collector processes
+    auto scint = std::make_unique<G4Scintillation>();
+#if G4VERSION_NUMBER < 1070
+    scint->SetStackPhotons(options_.scint_stack_photons);
+    scint->SetTrackSecondariesFirst(options_.scint_track_secondaries_first);
+    scint->SetScintillationByParticleType(options_.scint_by_particle_type);
+    scint->SetFiniteRiseTime(options_.scint_finite_rise_time);
+    scint->SetScintillationTrackInfo(options_.scint_track_info);
+    // These two are not in 10.7 and newer, but defaults should be sufficient
+    // for now
+    // scint->SetScintillationYieldFactor(fYieldFactor);
```

Comment:
Wait, so we're keeping these commented for now as a reminder to set them up later on? Is the scintillation yield factor here the same as, on newer G4 versions, the scintillation yield that is set via the material properties table?

Reply:
Possibly, I think @sethrj suggested that we focus on 10.7 and newer and revisit here if and only if experiments positively require support for the earlier versions. I'd have to look through the code to check on the yield being the same.